toolkit/tools/pkg/isomakerlib/isomaker.go

```diff
@@ -32,22 +33,23 @@ const (
-	config             configuration.Config // Configuration for the built ISO image and its installer.
-	configSubDirNumber int                  // Current number for the subdirectories storing files mentioned in the config.
-	baseDirPath        string               // Base directory for config's relative paths.
-	buildDirPath       string               // Path to the temporary build directory.
-	efiBootImgPath     string               // Path to the efiboot.img file needed to boot the ISO installer.
-	fetchedRepoDirPath string               // Path to the directory containing an RPM repository with all packages required by the ISO installer.
-	initrdPath         string               // Path to ISO's initrd file.
-	grubCfgPath        string               // Path to ISO's grub.cfg file. If provided, overrides the grub.cfg from the resourcesDirPath location.
-	outputDirPath      string               // Path to the output ISO directory.
-	releaseVersion     string               // Current Mariner release version.
-	resourcesDirPath   string               // Path to the 'resources' directory.
-	imageNameBase      string               // Base name of the ISO to generate (no path, and no file extension).
-	imageNameTag       string               // Optional user-supplied tag appended to the generated ISO's name.
+	enableBiosBoot     bool                    // Flag deciding whether to include BIOS bootloaders or not in the generated ISO image.
+	enableRpmRepo      bool                    // Flag deciding whether to include the contents of the Rpm repo folder in the generated ISO image.
+	unattendedInstall  bool                    // Flag deciding if the installer should run in unattended mode.
+	config             configuration.Config    // Configuration for the built ISO image and its installer.
+	configSubDirNumber int                     // Current number for the subdirectories storing files mentioned in the config.
+	baseDirPath        string                  // Base directory for config's relative paths.
+	buildDirPath       string                  // Path to the temporary build directory.
+	efiBootImgPath     string                  // Path to the efiboot.img file needed to boot the ISO installer.
+	fetchedRepoDirPath string                  // Path to the directory containing an RPM repository with all packages required by the ISO installer.
+	initrdPath         string                  // Path to ISO's initrd file.
+	grubCfgPath        string                  // Path to ISO's grub.cfg file. If provided, overrides the grub.cfg from the resourcesDirPath location.
+	outputDirPath      string                  // Path to the output ISO directory.
+	releaseVersion     string                  // Current Mariner release version.
+	resourcesDirPath   string                  // Path to the 'resources' directory.
+	additionalIsoFiles []safechroot.FileToCopy // Additional files to copy to the ISO media (absolute-source-path -> iso-root-relative-path).
```

Comment:
- `FileToCopy` should not be in chroot. It's really not specific to chroot at all.
- We shouldn't change isomake library to take any dependency on MIC specific types. `FileToCopy` is already using in the toolkit, so, it made sense to use it.
I think what we may do in the future is to make `FileToCopy` and its copy function (`addFilesToDestination`) into their own 'internal' go file for users to include... but that would cause a lot of modifications across the toolkit.
For this change, I think we'll leave `FileToCopy` usage...